packages/twenty-front/src/modules/workflow/workflow-steps/workflow-actions/delay-actions/components/WorkflowEditActionDelay.tsx

```diff
@@ -0,0 +1,122 @@
+import { SidePanelHeader } from '@/command-menu/components/SidePanelHeader';
+import { FormDateTimeFieldInput } from '@/object-record/record-field/ui/form-types/components/FormDateTimeFieldInput';
+import { Select } from '@/ui/input/components/Select';
+import { type WorkflowDelayAction } from '@/workflow/types/Workflow';
+import { WorkflowActionFooter } from '@/workflow/workflow-steps/components/WorkflowActionFooter';
+import { WorkflowStepBody } from '@/workflow/workflow-steps/components/WorkflowStepBody';
+import { useWorkflowActionHeader } from '@/workflow/workflow-steps/workflow-actions/hooks/useWorkflowActionHeader';
+import { WorkflowVariablePicker } from '@/workflow/workflow-variables/components/WorkflowVariablePicker';
+import { useEffect, useState } from 'react';
+import { HorizontalSeparator, IconCalendar } from 'twenty-ui/display';
+import { type SelectOption } from 'twenty-ui/input';
+import { useDebouncedCallback } from 'use-debounce';
+
+type WorkflowEditActionDelayProps = {
+  action: WorkflowDelayAction;
+  actionOptions:
+    | {
+        readonly: true;
+      }
+    | {
+        readonly?: false;
+        onActionUpdate: (action: WorkflowDelayAction) => void;
+      };
+};
+
+type DelayFormData = {
+  scheduledDateTime: string | null;
+};
+
+export const WorkflowEditActionDelay = ({
+  action,
+  actionOptions,
+}: WorkflowEditActionDelayProps) => {
+  const { headerTitle, headerIcon, headerIconColor, headerType, getIcon } =
+    useWorkflowActionHeader({
+      action,
+      defaultTitle: 'Delay',
+    });
+
+  const [formData, setFormData] = useState<DelayFormData>({
+    scheduledDateTime: action.settings.input.scheduledDateTime ?? null,
+  });
+
+  const delayOptions: Array<SelectOption<string>> = [
+    {
+      label: 'At a specific date or time',
+      value: 'specific_date_time',
+      Icon: IconCalendar,
+    },
+  ];
+
+  const saveAction = useDebouncedCallback(async (formData: DelayFormData) => {
+    if (actionOptions.readonly === true) {
+      return;
+    }
+    actionOptions.onActionUpdate({
+      ...action,
+      settings: {
+        ...action.settings,
+        input: {
+          scheduledDateTime: formData.scheduledDateTime,
+        },
+      },
+    });
+  }, 1_000);
+
+  useEffect(() => {
+    saveAction(formData);
```

Comment:
Is this what is done in other components? There is a useEffect for saveAction.flush but not for this? Let's try to stick to existing patterns